src/annotator/config/index.js

```diff
@@ -1,14 +1,84 @@
 import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
+/**
+ * List of allowed configuration keys per application context. Keys omitted
+ * in a given context will be removed from the relative configs when calling
+ * getConfig.
+ *
+ * @param {'sidebar'|'notebook'|'annotator'|'all'} [appContext] - The name of the app.
+ */
+function configurationContexts(appContext) {
+  switch (appContext) {
+    case 'annotator':
+      return ['clientUrl', 'showHighlights', 'subFrameIdentifier'];
+    case 'sidebar':
+      return [
+        'annotations',
+        'branding',
+        'enableExperimentalNewNoteButton',
+        'externalContainerSelector',
+        'focus',
+        'group',
+        'onLayoutChange',
+        'openSidebar',
+        'requestConfigFromFrame',
+        'services',
+        'showHighlights',
+        'sidebarAppUrl',
+        'theme',
+        'usernameUrl',
+      ];
+    case 'notebook':
+      return [
+        'branding',
+        'enableExperimentalNewNoteButton',
+        'focus',
+        'group',
+        'notebookAppUrl',
+        'requestConfigFromFrame',
+        'services',
+        'showHighlights',
+        'theme',
+        'usernameUrl',
+      ];
+    case 'all':
+    default:
+      // Complete list of configuration keys used for testing.
+      return [
+        'annotations',
+        'assetRoot',
+        'branding',
+        'clientUrl',
+        'enableExperimentalNewNoteButton',
+        'experimental',
+        'externalContainerSelector',
+        'group',
+        'focus',
+        'notebookAppUrl',
+        'onLayoutChange',
+        'openSidebar',
+        'query',
+        'requestConfigFromFrame',
+        'services',
+        'showHighlights',
+        'sidebarAppUrl',
+        'subFrameIdentifier',
+        'theme',
+        'usernameUrl',
+      ];
```

Comment:
Sounds good. I did this a bit different since I tried to keep it all in one function. I also don't see the need for Set() since the overlapping keys will cancel out, but please let me know if i missed something obvious or you intended for Set to improve readability.

Reply:
Hi Kyle - I used a `Set` to ensure that the keys were unique and avoid potential issues with upstream code that didn't handle duplicates properly. If it isn't a problem to have duplicates with the current upstream code then you can get away without it, but you probably want to note this assumption explicitly.